### PR TITLE
SMTP Setting

### PR DESCRIPTION
**Pull request by ammopt**

### CODE REVIEW

File: code/web/sys/Email/SMTPSetting.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+				'type' => 'label',
+				'label' => 'Id',
+				'description' => 'The unique id',
+			],
+			'name' => [
+				'property' => 'name',
+				'type' => 'text',
+				'label' => 'Server name',
+				'description' => 'The name of the server',
+				'default' => 'heyo',
```
Comment (mdnoble73) [inline code/web/sys/Email/SMTPSetting.php]: Let's remove the default value here

File: code/web/sys/Email/SMTPSetting.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		$mail->SMTPAuth = true;
+		$mail->Username = $this->user_name;
+		$mail->Password = $this->password;
+
+		if($this->ssl_mode != 'disabled'){
+			$mail->SMTPSecure = $this->ssl_mode;
+		}
+
+		$mail->From = $this->from_address;
+		$mail->FromName = 'Aspen Discovery';
```
Comment (mdnoble73) [inline code/web/sys/Email/SMTPSetting.php]: Should the From name be customizable? I could see libraries wanting to change that.

File: code/web/sys/Email/SMTPSetting.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		$mail->addAddress($to);
```
Comment (mdnoble73) [inline code/web/sys/Email/SMTPSetting.php]: We have a few cases where we send to multiple email addresses separated by semi-colons.  We should make sure we support that.

File: code/web/sys/Email/SMTPSetting.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		$mail->Host = $this->host;
+		$mail->SMTPAuth = true;
+		$mail->Username = $this->user_name;
+		$mail->Password = $this->password;
+
+		if($this->ssl_mode != 'disabled'){
+			$mail->SMTPSecure = $this->ssl_mode;
+		}
+
+		$mail->From = $this->from_address;
```
Comment (mdnoble73) [inline code/web/sys/Email/SMTPSetting.php:113]: We have a few cases where we override the from address in the other Email methods.

**Review by mdnoble73 — CHANGES_REQUESTED** [review]
Looks great.  Just a few little functionality changes in the sending to look at.

File: code/web/sys/Email/SMTPSetting.php (hunk 1 above)
Comment (ammopt) [reply code/web/sys/Email/SMTPSetting.php]: Addressed in https://github.com/Aspen-Discovery/aspen-discovery/pull/1850/commits/abf2b2fffb84394cd015b7ffa3ae09ac1fa5eeae

File: code/web/sys/Email/SMTPSetting.php (hunk 2 above)
Comment (ammopt) [reply code/web/sys/Email/SMTPSetting.php]: Addressed in https://github.com/Aspen-Discovery/aspen-discovery/pull/1850/commits/904c78302c1a37ba72180bfba0a43ca95700e0c4

File: code/web/sys/Email/SMTPSetting.php (hunk 3 above)
Comment (ammopt) [reply code/web/sys/Email/SMTPSetting.php]: Addressed in https://github.com/Aspen-Discovery/aspen-discovery/pull/1850/commits/19b0e48f97022ce09818dfec67727cc01af48108

File: code/web/sys/Email/SMTPSetting.php (hunk 4 above)
Comment (ammopt) [reply code/web/sys/Email/SMTPSetting.php:113]: Thank you so much for your review, Mark. Apologies for some of the 'basic' mishaps!
I don't fully understand what is required here though, can you please explain further?

From what I see in the other Email methods, they both utilize the configured 'from' value.
SendGrid:
```
$apiBody->from = new stdClass();
$apiBody->from->email = $sendGridSettings->fromAddress;
```

Amazon:
`$sesMessage->setFrom($this->fromAddress);`

Comment (mdnoble73) [reply code/web/sys/Email/SMTPSetting.php:113]: My apologies, I misremembered that

**Review by mdnoble73 — APPROVED** [review]

**Pull request merged**